### PR TITLE
Simply Refactor PatTycker

### PR DESCRIPTION
**Pull request by HoshinoTented**

### CODE REVIEW

File: base/src/main/java/org/aya/tyck/pat/PatternTycker.java [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+ * A Pattern Tycker for only one use.
+ */
+public final class PatternTycker {
+  public final @NotNull ExprTycker exprTycker;
+  public final @NotNull TypedSubst bodySubst;
+  private final @NotNull TypedSubst sigSubst = new TypedSubst();
+  private @NotNull Def.Signature<?> signature;
+  private @NotNull SeqView<Arg<Pattern>> patterns;
+  private final @NotNull MutableList<Arg<Pat>> wellTyped = MutableList.create();
+  private Term.@UnknownNullability Param currParam = null;
```
Comment (HoshinoTented) [inline base/src/main/java/org/aya/tyck/pat/PatternTycker.java:50]: I am not sure about this: `private Term.@UnknownNullability Param currParam = null;`

Comment (ice1000) [reply base/src/main/java/org/aya/tyck/pat/PatternTycker.java:50]: What specifically are you not sure about?

**Comment by ice1000** [comment]
Looks good in general

File: base/src/main/java/org/aya/tyck/pat/PatternTycker.java (hunk 1 above)
Comment (HoshinoTented) [reply base/src/main/java/org/aya/tyck/pat/PatternTycker.java:50]: Maybe we should pass `currParam` by argument? idk!

Comment (ice1000) [reply base/src/main/java/org/aya/tyck/pat/PatternTycker.java:50]: A mutable field is fine. That's the point of OOP! :)

**Pull request merged**